velox/type/Conversions.h

```diff
@@ -103,11 +103,11 @@ struct Converter<
       bool negative = false;
       // Setting decimalPoint flag
       bool decimalPoint = false;
-      if (v[0] == '-') {
+      if (v[0] == '-' || v[0] == '+') {
         if (len == 1) {
           VELOX_USER_FAIL("Cannot cast an '-' string to an integral value.");
         }
-        negative = true;
+        negative = v[0] == '-';
```

Comment:
This is to fix truncate cast used by Spark. Before this fix, below test fails.
> testCast<std::string, int8_t>("tinyint", {"+1"}, {1});

Reply:
Would it be possible to extract this fix into a separate PR?

Reply:
Yes, opened https://github.com/facebookincubator/velox/pull/8064 for this fix.